src/components/common/Toast/Toast.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { lazy, Suspense, useEffect } from 'react'
+import { useCart } from 'src/sdk/cart/useCart'
+import { useUI } from 'src/sdk/ui'
+
+const UIToast = lazy(() => import('src/components/ui/Toast'))
+
+function Toast() {
+  const { toasts, pushToast } = useUI()
+  const { messages } = useCart()
```

Comment:
We can move somewhere else. However, I think it makes sense to gather all notifications in one single place. This toast component is currently only used for displaying cart messages, but can eventually be used to display "online/offline" info, push notifications etc. By placing all notifications in a single place, makes it easier to debug messages and understand all messages that can be displayed by the Toast component.

I think a future PR can remove this `toasts` and `pushToast`  form `useUI` hook and make an inline implementation in here

Reply:
I understand that the cart notifications can be used in this case because we have the `useCart` hook and we can call it anywhere we want. But how will we be able to send notifications from other components that don't have a hook like this one?

Reply:
Usually notifications come from datasources, and usually datasources have a provider with hooks, right? Not sure